smart_open/smart_open_lib.py

```diff
@@ -322,8 +322,17 @@ def _shortcut_open(uri, mode, kwargs):
         return None
 
     open_kwargs = {}
-    if kwargs.errors is not None:
+    if kwargs.errors is not None and 'b' not in mode:
+        #
+        # Ignore errors parameter if we're reading binary.  This logic is
+        # necessary because:
+        #
+        #   - open complains if you pass non-null errors in binary mode
+        #   - our default mode is rb (incompatible with default Python open)
```

Comment:
How about we change this to be compatible?

Or at least make the mode mandatory, so people have to make a conscious decision between binary/text.

Reply:
Yes, we will do this in the next major release.

Reply:
BTW, we cannot make the mode mandatory, as that breaks compatibility with the built-in open.

We will switch to reading text mode as default in the next release.